### PR TITLE
RFC: Adding TX2i (P3849) support

### PR DESCRIPTION
**Pull request by compenguy**
I'm getting ready to test out these changes on hardware, but I wanted to post this for feedback first, in case I'm missing anything important.

I most _concerned_ about getting the values in `jetson-tx2i.conf` and `flashvars` correct enough before I try flashing, but I think the most interesting discussions will probably be about my changes to `tegra186-flash-helper.sh`.

### CODE REVIEW

File: recipes-bsp/tegra-binaries/files/tegra186-flash-helper.sh [hunk 1]
```diff
@@ -32,37 +32,65 @@ if [ -z "$FLASHVARS" ]; then
...
+            PMICREV="c04"
+            ;;
+        B00)
+            ;;
+        *)
+            echo "ERR: unsupported board revision: $boardrev" >&2
+            exit 1
+            ;;
+    esac
+elif [ "$boardid" = "3849" ]; then
```
Comment (madisongh) [inline recipes-bsp/tegra-binaries/files/tegra186-flash-helper.sh]: `3489`

File: recipes-bsp/tegra-binaries/files/tegra186-flash-helper.sh [hunk 2]
```diff
@@ -32,37 +32,65 @@ if [ -z "$FLASHVARS" ]; then
...
+    BOARDREV="a00"
+    PMICREV="a00"
+    BPFDTBREV="a00";
+    if [ "${boardrev}" = "A02" ]; then
```
Comment (madisongh) [inline recipes-bsp/tegra-binaries/files/tegra186-flash-helper.sh]: If I'm reading the `process_board_version` function in the L4T kit correctly, this check should be for `${boardrev}` starting with 0, 1, or 2 (i.e., less than 300).

**Comment by compenguy** [comment]
I will apply these fixes, then try it out on actual hardware.  If everything looks good, I'll update this pull request, rebase the changes on master and submit a pull request for it, too.

**Comment by madisongh** [comment]
Were you able to flash your TX2i with these changes in place? If this is ready, I'll prep for pulling them in.

**Comment by compenguy** [comment]
I was getting ready to swap in a TX2i when I realized I didn't have a heatsink for it, so I have one one order now, and I'll try the TX2i out as soon as I have my heatsink for it.

**Comment by madisongh** [comment]
@compenguy Thanks for the update

**Comment by compenguy** [comment]
Tested the tip of my branch on a TX2i - it flashed, booted up, and I was able to verify ECC memory operation by checking DRAM.

**Comment by madisongh** [comment]
I manually merged this with a squash commit into both master and warrior.  Thanks!

**Pull request closed without merging**